File: backend/src/main/java/com/jimuqu/devops/entity/Host.java
package com.jimuqu.devops.entity;

import com.jimuqu.devops.common.entity.BaseEntity;
import com.jimuqu.devops.common.enums.HostStatus;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 主机管理实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Host extends BaseEntity {
    
    /**
     * 主机名称
     */
    private String name;
    
    /**
     * 主机IP地址
     */
    private String hostIp;
    
    /**
     * SSH端口
     */
    private Integer port = 22;
    
    /**
     * SSH用户名
     */
    private String username;
    
    /**
     * SSH密码(加密存储)
     */
    private String password;
    
    /**
     * SSH私钥
     */
    private String privateKey;
    
    /**
     * 主机状态
     */
    private HostStatus status = HostStatus.OFFLINE;
    
    /**
     * 主机描述
     */
    private String description;
}

File: frontend/src/types/host.ts
// 主机分组类型定义
export interface HostGroup {
  id: string;
  name: string;
  description?: string;
  sort: number;
  hostCount: number;
  createdAt: string;
  updatedAt: string;
}

// 主机信息类型定义
export interface Host {
  id: string;
  groupId: string;
  groupName: string;
  name: string;
  ip: string;
  port: number;
  username: string;
  password: string;
  authType: 'password' | 'key';
  privateKey?: string;
  description?: string;
  status: 'online' | 'offline' | 'unknown';
  lastConnected?: string;
  createdAt: string;
  updatedAt: string;
}

// 主机分组表单类型
export interface HostGroupForm {
  name: string;
  description?: string;
}

// 主机信息表单类型
export interface HostForm {
  groupId: string;
  name: string;
  ip: string;
  port: number;
  username: string;
  password: string;
  authType: 'password' | 'key';
  privateKey?: string;
  description?: string;
}

// API 响应类型
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
  success: boolean;
}

// 分页响应类型
export interface PageResponse<T> {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}